pkg/provisioner/crdprovisioner.go

```diff
@@ -347,6 +347,14 @@ func (c *CrdProvisioner) PublishVolume(
 		if azVolumeAttachmentInstance.Status.Detail != nil && azVolumeAttachmentInstance.Status.Detail.PublishContext != nil && azVolumeAttachmentInstance.Status.Detail.Role == v1alpha1.PrimaryRole {
 			return azVolumeAttachmentInstance.Status.Detail.PublishContext, nil
 		}
+
+		// If attachment is pending or CRI is scheduled for deletion, retry until operations are complete
+		if azVolumeAttachmentInstance.Status.State == v1alpha1.Attaching {
+			return nil, status.Errorf(codes.Internal, "attachment still in process for volume (%s) and node (%s)", volumeName, nodeID)
+		} else if azVolumeAttachmentInstance.DeletionTimestamp != nil {
+			return nil, status.Errorf(codes.Internal, "need to wait until attachment is fully deleted for volume (%s) and node (%s) before attaching", volumeName, nodeID)
+		}
```

Comment:
Are these redundant to/superceded by the changes in #175?

Reply:
Ah yes. These changes are redundant and need to be removed.